src/Parsers/IAST.h

```diff
@@ -78,11 +78,13 @@ class IAST : public std::enable_shared_from_this<IAST>, public TypePromotion<IAS
     virtual ASTPtr clone() const = 0;
 
     /** Get hash code, identifying this element and its subtree.
+     *  Hashing by default ignores aliases (e.g. identifier aliases, function aliases, literal aliases) which is
+     *  useful for common subexpression elimination. Set 'ignore_aliases = false' if you don't want that behavior.
       */
     using Hash = CityHash_v1_0_2::uint128;
-    Hash getTreeHash() const;
-    void updateTreeHash(SipHash & hash_state) const;
-    virtual void updateTreeHashImpl(SipHash & hash_state) const;
+    Hash getTreeHash(bool ignore_aliases = true) const;
```

Comment:
One nitpick - I generally tend to avoid default arguments for functions because:

1. It is error-prone, and it was proven by a few bugs in our codebase.
2. Default arguments in virtual functions increase the size of the virtual table.
3. I've heard that Rust does not have default function arguments, so it might be good.